astropy/cosmology/flrw.py

```diff
@@ -185,10 +170,10 @@ def __init__(self, H0, Om0, Ode0, Tcmb0=0.0*u.K, Neff=3.04, m_nu=0.0*u.eV,
         # neutrinos)
         self._massivenu = False
         if self._nneutrinos > 0 and self._Tcmb0.value > 0:
-            self._neff_per_nu = self._Neff / self._nneutrinos
+            # not set yet, just validated
+            m_nu = _validate_with_unit(self, self.__class__.m_nu, m_nu)
 
-            with u.add_enabled_equivalencies(cls.m_nu.equivalencies):
-                m_nu = m_nu << cls.m_nu.unit
+            self._neff_per_nu = self._Neff / self._nneutrinos
 
             # Now, figure out if we have massive neutrinos to deal with,
```

Comment:
I'm still a bit confused about the logic here -- could all this logic be moved to the validator, thus avoiding the assignment at the end? Alternatively, might it make more sense to *use* the simple validator above and just update the private attribute at the end?

Reply:
I think the latter is the better move for now. In this PR I refrained from assigning any attributes in the validator because I'm still unsure if that's a good code pattern.
E.g. 

```python
@H0.validator
def H0(self, param, value):
    value = _validate_with_unit(self, param, value)

    self._h = value.value / 100.0
    self._hubble_distance = (const.c / value).to(u.Mpc)
    H0_s = value.value * H0units_to_invs
    self._hubble_time = u.Quantity(sec_to_Gyr / H0_s, u.Gyr)

    return value
```

Similarly a lot of the auxiliary neutrino attributes could be moved into the ``m_nu.validator``.
But I think that's good for a v5.1 followup.

Reply:
OK, to be honest, I prefer the simpler update-the-private attribute later too. And nothing we do here affects the API of `Parameter`, which suggests we may have converged on that!

Reply:
Just to be sure, are you thinking of doing the update-the-private-attribute in this PR still? I think it has the advantage of leaving things in a consistent state, in which there is no need to use the nominally private validation functions.

Reply:
Yes. The commit that I most recently pushed moved the contents of the ``m_nu`` validator into ``FLRW.__init__``. Now only ``Ob0`` has a non pre-registered validator.

Reply:
Ah, now I see how it works... continuing to review...